src/main/java/core/basesyntax/HelloWorld.java

```diff
@@ -4,5 +4,20 @@
  * Feel free to remove this class and create your own.
  */
 public class HelloWorld {
+    public static void main(String[] args) {
 
+        Figure[] array = new Figure[6];
+        FigureSupplier supplier = new FigureSupplier();
+        for (int i = 0; i < array.length / 2; i++) {
+            array[i] = supplier.getRandomFigure();
+        }
+
+        for (int i = array.length / 2; i < array.length; i++) {
+            array[i] = supplier.getDefault();
+        }
```

Comment:
```suggestion
        for (int i = 0; i < array.length / 2; i++) {
            array[i] = supplier.getRandomFigure();
        }

        for (int i = array.length / 2; i < array.length; i++) {
            array[i] = supplier.getDefault();
        }
```

use one loop

Reply:
![image](https://github.com/mate-academy/jv-oop-advanced/assets/94477184/5bd1336b-328c-4a5a-ac61-ffec5e3ed211)
  can't use one loop, the condition is always true

Reply:
You need if else, not do while.